R/paginate_listing.R

```diff
@@ -58,9 +58,17 @@ paginate_listing <- function(lsting, lpp = 15,
   if (flines > 0) {
     flines <- flines + dheight + 1L
   }
-  ## row lines per page
-  rlpp <- lpp - cinfo_lines - tlines - flines
-  pagdf <- make_row_df(lsting, colwidths)
+
+  if (!is.null(lpp)) {
+    ## row lines per page
+    rlpp <- lpp - cinfo_lines - tlines - flines
+
+    pagdf <- make_row_df(lsting, colwidths)
+  } else {
+    pagdf <- make_row_df(lsting, colwidths)
+    rlpp <- nrow(pagdf)
```

Comment:
this isn't quite right, as rows can take up more than one line (thought they usually shouldn't in a listing, it is supported).

add back in the `pagdf` variable creation and then set `rlpp` to the sum of `pagdf$self_extent + pagdf$nreflines`. That should be both correct now and remain correct once we add referential footnote support.

Reply:
this above doesn't seem to work because there is a pagdf$self_extent value for each row. rlpp should be a single value that is passed to `pag_indices_inner`. I'll look for a way to capture self_extent as well the rows per page.

![image](https://user-images.githubusercontent.com/73252787/206209163-311d1189-c2cb-4a6f-be54-f63578f4fe00.png)

Reply:
Right, you need to sum it before the subtraction.